spacy_llm/models/hf/base.py

```diff
@@ -48,34 +52,45 @@ def __call__(self, prompts: Iterable[_PromptType]) -> Iterable[_ResponseType]:
         RETURNS (Iterable[_ResponseType]): API responses.
         """
 
-    @property
+    def _check_model(self) -> None:
+        """Checks whether model is supported. Raises if it isn't."""
+        if (
+            self._name not in self.get_model_names()
+            and self._name.replace(f"{self.get_hf_account()}/", "")
+            not in self.get_model_names()
+        ):
+            raise ValueError(
+                f"Model '{self._name}' is not supported - select one of {self.get_model_names()} instead"
+            )
+
+    @classmethod
+    def get_model_names(cls) -> Tuple[str, ...]:
+        """Names of supported models for this HF model implementation.
+        RETURNS (Tuple[str]): Names of supported models.
+        """
+        return tuple(str(arg) for arg in cls.MODEL_NAMES.__args__)  # type: ignore[attr-defined]
+
+    @staticmethod
     @abc.abstractmethod
-    def supported_models(self) -> Iterable[str]:
-        """Get names of supported models.
-        RETURNS (Iterable[str]): Names of supported models.
+    def get_hf_account() -> str:
```

Comment:
We should probably make this a property?
Something like 
```
@property
@abstractmethod
def hf_account(self):
```

Reply:
Done in https://github.com/explosion/spacy-llm/pull/176/commits/8d6ae3fa2c6246ad18554862fe586a6a3c1c2b28.